feat: Migrate Docs to Docusaurus

adiati98 opened the pull request:
## Linked Issue

Closes #454 

<!--

If you have a pull request related to a current issue please link to that issue number.

That issue can be linked to the pull request by using the side panel in the Github UI or using the `#` symbol followed by the number of the associated issue.

To link a pull request to an issue to show that a fix is in progress and to automatically close the issue when someone merges the pull request, type the keyword "Closes" followed by a reference to the issue. For example, Closes #404 or Closes Virtual-Coffee/virtualcoffee.io/issues/404.

-->

## Description

This PR moves our documentation to Docusaurus for easy access and better maintenance. It holds below changes:

- Install and use latest version of Docusaurus
- Use `pnpm` for consistency with vcio website
- Move all documentation to the `docs` folder
- Restructure and rename some files and folders for consistency (previously some folders and files used capital letters and underscores) and clarity
- Change some `h1` to clarify the contents
- Remove Table of Contents (TOC) as it's provided by Docusaurus
- Change GitHub to Docusaurus admonitions

### Screen Recording

![vc-docs-docusaurus](https://github.com/user-attachments/assets/95297434-b5fb-4f47-bd80-dfe6f60ac1c2)


## How to Test Locally

1. Pull this branch (`feat/migrate-to-docusaurus`) to your local machine
2. Run `pnpm install` to install dependencies
3. Run `pnpm start` to run Docusaurus

## ⚠️  Special Notes

- This PR doesn't change any content of existing docs
- The README in the `docs` folder is a duplication from existing README with adjusted links
- The keywords in the front matter are generated and haven't been adjusted
- The old docs are included in the comment in the `sidebar.js` in case we need it. 

We will create separate issues for any changes regarding the above matters to make sure this PR is not going out of the issue's scope.

<!--

A pull request description describes what constitutes the Pull Request and what changes you have made to the code.

It explains what you've done, including any code changes, configuration changes, migrations included, new APIs introduced, changes made to old APIs, any new workers/crons introduced in the system, copy changes, and so on. You get the gist.

A good description informs everyone that is reaading it of the purpose of the pull request. This helps not just the current maintainers but anyone reading it now or in the future to understand your intent.

If the request is not complete but you want feedback use  Draft Pull Request option of the Pull request dropdown menu.

@mention individuals that you want to review the PR, and mention why. (“ @username I want to know what you think of this code.”)

-->

## Methodology

<!--

This section explains why the above changes explained were done.

Sometimes a developer feels that it's okay to write "Business/Product requirement" in the description. That's fine, but doing so defeats the purpose of this section.

If there is a better explanation as to why the changes were suggested, it's always good to attach a document reference link for that information.

A good "Why" section should explain the reasoning behind any changes.

-->

## Code of Conduct

> By submitting this pull request, you agree to follow our [Code of Conduct](https://virtualcoffee.io/code-of-conduct/)

BekahHW commented:
One thing we might want to look at is adding directions on how to run the repo in the ReadMe.

adiati98 commented:
> One thing we might want to look at is adding directions on how to run the repo in the ReadMe.

I was thinking about adding this once this PR is merged because I want this PR to only move our docs to Docusaurus as is. I'll create a new issue for this! 😊

adiati98 commented:
> Another thing to note is that docusaurus is case sensitive. I've updated it, but wanted to make sure you knew.
> 
> ![image](https://github.com/user-attachments/assets/66e67ecf-dc2b-4ff0-b725-f9df6e5a4e10)
> 

Hmm... This is weird. I didn't get these errors when I ran the project locally. 😅

I thought I've renamed all folders and files to be lowercase for consistency. I might have missed these! Thanks for spotting them!
